### PR TITLE
fix: emulate `event.platform` during development even if route is undefined

### LINKED ISSUE #11996: adapter-cloudflare emulated bindings not always available
### Describe the bug

My work flow was using miniflare to polyfill the cloudflare platform environment in handle server hooks for local development.

_For some of the people may concern it's easy to use wrangler-proxy or cfw-bindings-wrangler-bridge for remote data. Combining with import.meta.env.MODE it's very convenient to switch between dev, preview, live mode for local and remote data._

In the recent adapter update it's doing mostly what miniflare did. I got 2 projects 1 using authjs and another using lucia auth.
In the lucia auth project I can remove the manually implemented miniflare part and rely on the adapter for local development. For remote data the bridge is still needed.
In the auth js project it's broken as soon as I try to login. It's complaining about platform is undefined and it's trying to access platform.env. Those are implemted in handle server hooks, not sevltekit endpoints.

The lucia auth project is facing the same if I try to move the /google and /google/callback sveltekit endpoints into handle server hooks. It'll complain about platform is undefined.

I'm not familiar to how and when adapter inject the platform into the event object. From my simple testing it seems to inject the platform only when there is +server.ts as a sveltekit endpoint with known route. And no platform is injected when the endpoints logic is implemented in the handle server hooks directly.

### Reproduction

Not available in near future

### Logs

_No response_

### System Info

```Shell
-
```

### Severity

annoyance

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by yutak23**
This ought to fix #11996

Only if `DEV`, config is an empty object and prerender is set to `false` so that `event.platform` can be emulated.
I believe this will improve the development experience by preventing 500 errors and allowing development to occur in a more local development environment similar to the production environment.

---

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### Edits

- [x] Please ensure that 'Allow edits from maintainers' is checked. PRs without this option may be closed.

### CODE REVIEW

File: .changeset/shiny-suns-scream.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+In dev, emulate event.platform even if route is undefined
```
Comment (eltigerchino) [inline .changeset/shiny-suns-scream.md:5]:
```suggestion
fix: emulate `event.platform` during development even if route is undefined
```

File: packages/kit/src/runtime/server/respond.js [hunk 2]
```diff
@@ -297,6 +297,10 @@ export async function respond(request, options, manifest, state) {
 			}
 		}
 
+		if (!event.platform && DEV && state.emulator?.platform) {
+			event.platform = await state.emulator.platform({ config: {}, prerender: false });
+		}
```
Comment (eltigerchino) [inline packages/kit/src/runtime/server/respond.js:302]: @benmccann is this what you meant in https://github.com/sveltejs/kit/issues/11996#issuecomment-2211792494 ? I suppose one more check we need to add is if we're currently rendering a fallback

**Comment by codenoid** [comment]
Hi!, should this get merged?

**Comment by codenoid** [comment]
Hi!, should this get merged?

I manually change Kit's node_modules file, and this change really made it to work

File: packages/kit/src/runtime/server/respond.js [hunk 3]
```diff
@@ -297,6 +297,10 @@ export async function respond(request, options, manifest, state) {
 			}
 		}
 
+		if (!event.platform && DEV && state.emulator?.platform) {
+			event.platform = await state.emulator.platform({ config: {}, prerender: false });
```
Comment (eltigerchino) [inline packages/kit/src/runtime/server/respond.js:301]:
```suggestion
			event.platform = await state.emulator.platform({
				config: {},
				prerender: !!state.prerendering?.fallback
			});
```

File: .changeset/shiny-suns-scream.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+fix: emulate `event.platform` during development even if route is undefined
```
Comment (eltigerchino) [inline .changeset/shiny-suns-scream.md:5]:
```suggestion
fix: emulate `event.platform` even when the route does not exist
```

**Review by eltigerchino — APPROVED** [review]

**Pull request merged**